tests/run/i4364b.scala

```diff
@@ -0,0 +1,23 @@
+object Test {
```

Comment:
It is not clear what you are testing here that is not tested in `i4364a.scala`. I suggest you add a test to `tests/neg-custom-args/fatal-warnings` to test the warnings:
```scala
import java.io._

class Test {
  def foo(c: java.util.function.Consumer[String]) = ()

  val out1 = new ObjectOutputStream(x => println(x))
  val out2 = new ObjectOutputStream(println(_))
  val out3 = new ObjectOutputStream(println) // error: OutputStream is not @FunctionalInterface

  def f(x: String) = ()
  foo(f) // Ok: Consumer is @FunctionalInterface
}
```

Reply:
```scala
def f(x: Int): Unit = assert(false)
def f(x: String): Unit = assert(false)
def f: java.io.OutputStream = new java.io.OutputStream { def write(x: Int) = () }

val oosF = new java.io.ObjectOutputStream(f)
oosF.write(0)
```
Through the test, I tried to check that whether method overloading is correctly resolved when candidates exist without considering SAM type. In this case, the second `f` is valid choice when it is eta-expanded and `OutputStream` is considered as SAM type. However, the third `f` should be choosed. Therefore, I think this test needs to remain.

As you suggested, it would be better to check a warning in a seperate test. I'll remove
```scala
val oosG = new java.io.ObjectOutputStream(g)
oosG.write(0)
oosG.close()
```
from test b and add a test about a warning in `tests/neg-custom-args/fatal-warnings`.

Reply:
Good point. I suggest a simpler test though:
```scala
import java.util.function.Consumer

object Test {
  def foo(c: Consumer[String]) = c.accept("")

  def bar(x: String): Unit = ???
  def bar: Consumer[String] = new Consumer {  def accept(s: String) = () }

  def main(args: Array[String]) = {
    foo(bar)
  }
}
```

Reply:
I agree that this one is simpler and clearer. I'll fix it. Thank you!